### DISCUSSION TITLE
Font Bakery

### DISCUSSION THREAD

**Discussion by basil** [neanes/neanes#373] [Ideas]
Can this project benefit from [Font Bakery](https://github.com/fonttools/fontbakery)?

https://www.youtube.com/watch?v=vFjQEkJv9BQ

**Comment by danielgarthur** [comment]
While I do think there are some checks in the library that would benefit SBMuFL, I think it's geared toward an alphabetic script. For the martyria + fthora issue I wrote a script that printed out the distances. It's pretty easy to read the contours from the Font Forge SFD file and measure the distance to the anchor. I want to someday clean up the script and include it as part of a scripting toolkit for SBMuFL.

**Reply by basil** [reply]
> I think it's geared toward an alphabetic script

I got that impression as well, but maybe as a general framework it might be easier to write tests there than parsing a Font Forge SFD file directly. You never know, though, as sometimes frameworks can be bloated and inefficient compared to doing the task directly.

**Reply by basil** [reply]
I added Font Bakery to the build in https://github.com/neanes/sbmufl/pull/28. It should be possible to write a Font Bakery profile to enforce the invariants needed for a quality SBMuFL font, although I have no idea what those invariants are. See https://fontbakery.readthedocs.io/en/latest/developer/writing-profiles.html#writing-checks for the developer docs.
